src/mapgen/mapgen.cpp

```diff
@@ -238,7 +238,8 @@ u32 Mapgen::getBlockSeed(v3s16 p, s32 seed)
 
 u32 Mapgen::getBlockSeed2(v3s16 p, s32 seed)
 {
-	u32 n = 1619 * p.X + 31337 * p.Y + 52591 * p.Z + 1013 * seed;
+	// Unsigned magic seed prevents undefined behavior.
```

Comment:
```suggestion
	// Multiply by unsigned number to avoid signed overflow (UB)
```
could also attach an `U` to all constants on that line, wouldn't hurt